### PR TITLE
add tips contract

### PR DESCRIPTION
**Pull request by iavl**
### Checklist
- [x] I have clearly commented on all the main functions following the [NatSpec Format](https://docs.soliditylang.org/en/v0.8.0/natspec-format.html)
- [x] I have updated the abi and docs
- [x] I tested locally to make sure this feature/fix works

### CODE REVIEW

File: contracts/misc/Thanks.sol [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        address to = IERC721(web3Entry).ownerOf(toCharacterId);
+        // check
+        if (msg.sender != from) revert ErrCallerNotCharacterOwner();
+
+        // transfer token
+        IERC20(token).safeTransferFrom(from, to, amount);
+
+        // emit event
+        emit ThankCharacter(fromCharacterId, toCharacterId, token, amount);
+    }
```
[the remark this answers is not in the document]
Comment (iavl) [reply contracts/misc/Thanks.sol]: fixed

File: contracts/misc/Thanks.sol [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+        address to = IERC721(web3Entry).ownerOf(toCharacterId);
+        // check
+        if (msg.sender != from) revert ErrCallerNotCharacterOwner();
+
+        // transfer token
+        IERC20(token).safeTransferFrom(from, to, amount);
+
+        // emit event
+        emit ThankNote(fromCharacterId, toCharacterId, toNoteId, token, amount);
+    }
```
[the remark this answers is not in the document]
Comment (iavl) [reply contracts/misc/Thanks.sol]: fixed

File: contracts/misc/Thanks.sol [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+        uint256 indexed toNoteId,
+        address token,
+        uint256 amount
+    );
+
+    /**
+     * @notice Initialize the contract.
+     * @param web3Entry_ Address of web3Entry.
+     */
+    function initialize(address web3Entry_) external initializer {
```
[the remark this answers is not in the document]
Comment (iavl) [reply contracts/misc/Thanks.sol]: fixed

File: contracts/misc/Thanks.sol [hunk 4]
```diff
@@ -0,0 +1,121 @@
+// SPDX-License-Identifier: MIT
+// solhint-disable comprehensive-interface
+pragma solidity 0.8.10;
```
[the remark this answers is not in the document]
Comment (iavl) [reply contracts/misc/Thanks.sol]: fixed

**Review by Fooooooooooox — APPROVED** [review]

**Pull request merged**